libutils/file_lib.c

```diff
@@ -835,7 +835,7 @@ FILE *safe_fopen(const char *path, const char *mode, ...)
     }
     else
     {
-        fd = safe_open(path, flags);
+        fd = safe_open(path, flags, 0);
```

Comment:
LGTM thought it was good practice to be explicit about it, though I am still on the fence about whether we should use variadic functions at all.

Reply:
> I am still on the fence about whether we should use variadic functions at all.

I think it makes sense due to the standard `open()` doing the same. But let's see what @olehermanse thinks.

Reply:
@vpodzime @karlhto It makes sense to me that `safe_open()` accepts the same arguments as `open()`, and `safe_fopen()` matches `fopen()`. 

When it comes to the optional argument for `open()`, I really don't like it. I think I (slightly) prefer to have 2 functions, `safe_open()` taking 2 arguments only, and defaulting to restrictive `mode` argument on `O_CREAT`. `safe_open_mode()` (or similar name, `safe_open_perms()`) can have the required, third, `mode` parameter. The implementation of `safe_open()` would then call `safe_open_mode()` internally.

Reply:
@olehermanse I agree, seems like that's the best way to do it in C -- especially because it might seem kind of arbitrary in which situations you are supposed to use the optional argument.

I think that `safe_open` (and then in effect `safe_fopen`) should default to mode `0600`, and then we should have additional functions (like `safe_open_mode` and `safe_fopen_mode`) if you explicitly want to define a different mode.